scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/InMemoryConnectionStore.java

```diff
@@ -272,4 +294,31 @@ public void handshakeCompleted(final InetSocketAddress peer) {
 	public void handshakeFailed(final InetSocketAddress peer, Throwable error) {
 		// nothing to do
 	}
+
+	/**
+	 * Check if the connection is expired, if this is the case remove it from
+	 * the store.
+	 * 
+	 * @return return the connection only if the connection has no session
+	 *         established or if the established session is not expired.
+	 */
+	private Connection checkValidity(Connection connection) {
+		if (connection == null) {
+			return null;
+		}
+		if (!connection.hasEstablishedSession()) {
+			if (!connection.hasSessionTicket()) {
+				return connection;
+			} else if (!connection.getSessionTicket().isExpired(sessionLifeTime)) {
+				return connection;
+			}
+		} else if (!connection.getEstablishedSession().isExpired(sessionLifeTime)) {
+			return connection;
+		}
+
+		// connection expired
+		connections.remove(connection.getPeerAddress(), connection);
+		removeSessionFromCache(connection);
```

Comment:
`removeSessionFromCache` will not remove session from cache, if the connection is created for a ticket.
See comment above; I would prefer to process the expired tickets in `deepfind`.